app/src/main/java/com/example/mystore/ui/profile/ProfileScreen.kt

```diff
@@ -124,7 +131,18 @@ fun ProfileScreen(
                 enabled = true,
                 onClick = {
                     viewModel.logout {
-                        destinationsNavigator.navigate(LoginScreenDestination)
+                        val hasLoginDestination = destinationsNavigator.popBackStack(
```

Comment:
Great implementation! However, this block has grown quite big. I'd advise to extract the whole thing to a function outside of the view and then simply pass the function reference, like `viewModel.logout(onLogout = navigateToLogin)`, provided you name this function `navigateToLogin()`, of course.